### PR TITLE
Update libkv dependency

### LINKED ISSUE #926: letsEncrypt not working with etcd
When I try to configure Traefik in combination with 'etcd' as a backend store, i can't get LetsEncrypt working.

traefik.toml (imported with storeconfig)
```toml
[web]
address = ":8080"
defaultEntryPoints = ["https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
[acme]
email = "me@domain.tld"
storage= "/traefik/acme/account"
entryPoint = "https"
onDemand = false
OnHostRule = true
caServer = "https://acme-staging.api.letsencrypt.org/directory"
[[acme.domains]]
   main = "domain.tld"
   sans = ["www.domain.tld", "ww2.domain.tld"]
[etcd]
endpoint = "127.0.0.1:4001"
watch = true
prefix = "/traefik"
```

Output in the logs

```shell
Dec 02 09:23:18 core-01 rkt[21448]: [131075.886503] traefik[5]: time="2016-12-02T10:23:18+01:00" level=info msg="Skipping same configuration for provider etcd"
Dec 02 09:23:22 core-01 rkt[21448]: [131079.650257] traefik[5]: time="2016-12-02T10:23:22+01:00" level=debug msg="Challenge GetCertificate www.mydomain.tld"
Dec 02 09:23:22 core-01 rkt[21448]: [131079.650881] traefik[5]: time="2016-12-02T10:23:22+01:00" level=debug msg="Loading ACME certificates [www.mydomain.tld]..."
Dec 02 09:23:22 core-01 rkt[21448]: [131079.661423] traefik[5]: time="2016-12-02T10:23:22+01:00" level=debug msg="Challenge GetCertificate www.mydomain.tld"
Dec 02 09:23:22 core-01 rkt[21448]: [131079.662657] traefik[5]: time="2016-12-02T10:23:22+01:00" level=debug msg="Loading ACME certificates [www.mydomain.tld]..."
Dec 02 09:23:22 core-01 rkt[21448]: [131080.212813] traefik[5]: time="2016-12-02T10:23:22+01:00" level=debug msg="Challenge Present www.mydomain.tld"
Dec 02 09:23:22 core-01 rkt[21448]: [131080.238591] traefik[5]: time="2016-12-02T10:23:22+01:00" level=debug msg="Challenge Present www.mydomain.tld"
```

From the output in the logs I can't really find out what is exactly going on. I had a look at the source code to see where it might go wrong, but my Go knowledge is no-exiting atm. 

Slack reference : https://traefik.slack.com/archives/support/p1480674299005971

### PR DESCRIPTION
**Pull request by nmengin**
### What does this PR do?

In the way to migrate fix some bugs with KV store, essentially ETCDV3, this PR migrates Træfik to the abronan libkv fork we have to migrate first stært in the way to fix dependencies.

### Motivation

Fixes #926 #1034 #2069 #1974 #1975 

### More

- [x] Added example for manual tests

### Additional information

The new libkv version allows connections to ETCD through both `V2` **and** `V3` API.
In the way to avoid breaking change, the parameter `userAPIV3` was added to the backend ETCD.

**Note :** Both the Atcd API V2 and the option `userAPIV3` are **deprecated**. More information about deprecated option in proposal #2212.

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -36,13 +36,19 @@ import:
   - tlsconfig
 - package: github.com/docker/go-units
   version: 9e638d38cf6977a37a8ea0078f3ee75a7cdb2dd1
+- package: github.com/coreos/etcd
+  version: v3.2.9
```
[the remark this answers is not in the document]
Comment (nmengin) [reply glide.yaml]: Modification done in the docker-compose file.

File: docs/configuration/backends/etcd.md [hunk 2]
```diff
@@ -59,3 +69,7 @@ prefix = "/traefik"
 To enable constraints see [backend-specific constraints section](/configuration/commons/#backend-specific).
 
 Please refer to the [Key Value storage structure](/user-guide/kv-config/#key-value-storage-structure) section to get documentation on Traefik KV structure.
+
+!!! Note
+    the option `useAPIV3` allows using Etcd API V3 only if it's set to true.
+    This option will be **deprecated** and API V3 will be the only version compatible with Træfik.
```
Comment (emilevauge) [inline docs/configuration/backends/etcd.md]: s/will be/is

File: examples/cluster/manage_cluster_docker_environment.sh [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            down_environment
+            main --start $2
+            ;;
+        *)
+            show_usage && exit 6
+            ;;
+    esac
+}
+
+main $@
```
Comment (emilevauge) [inline examples/cluster/manage_cluster_docker_environment.sh]: 247 lines of fun ;)

vomit

Are we sure we want to maintain this in the repo ?

File: provider/etcd/etcd.go [hunk 4]
```diff
@@ -31,7 +34,13 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 
 // CreateStore creates the KV store
 func (p *Provider) CreateStore() (store.Store, error) {
-	p.SetStoreType(store.ETCD)
-	etcd.Register()
+	if p.UseAPIV3 {
+		etcdv3.Register()
+		p.SetStoreType(store.ETCDV3)
+	} else {
+		log.Warn("The ETCD API V2 is deprecated. Please use API V3 instead")
```
Comment (emilevauge) [inline provider/etcd/etcd.go]: Could you add a comment `// TODO: Deprecated` before ?

File: examples/cluster/manage_cluster_docker_environment.sh (hunk 3 above)
Comment (nmengin) [reply examples/cluster/manage_cluster_docker_environment.sh]: For the moment I prefer keep it as long as we do not have end-to-end test about the cluster mode.
It allows generating easily a Dockerized Træfik cluster...

Moreover, shell is life... shell is lake a unicorn...

whererainbow

**Pull request merged**